### PR TITLE
Update some dependencies

### PR DESCRIPTION
**Pull request by MikailBag**
Explicitly mark send_struct and recv_struct as unsafe

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -16,12 +16,12 @@ ser_json = ["serde", "serde_json"]
...
+nix = "0.13"
+error-chain = "0.12"
 serde = { version = "1.0", optional = true }
 serde_cbor = { version = "0", optional = true }
 serde_json = { version = "1.0", optional = true }
 bincode = { version = "1.0", optional = true }
 
 [dev-dependencies]
-shmemfdrs = "0"
+shmemfdrs = "0.1"
```
Comment (valpackett) [inline Cargo.toml:29]: don't add more specific versions if it's `0`

**Comment by valpackett** [comment]
ehhh… I don't want to require typing `unsafe {}` all the time. Even `std::mem::swap` is not `unsafe`.

**Comment by MikailBag** [comment]
1. Usually, when major version is 0, library authors bump minor version on breaking changes.
2. mem::swap swaps two valid (if one is invalid, then UB already happened) `T`s, so in output we have two correct values of type T. I don't see, how mem::swap can trigger UB.
3. Regarding too many unsafes. I think it can be solved by an unsafe trait ByteSequence (maybe, there is one already). T: ByteSequence should mean following:
- T: Clone
- T: !Drop and doesn't have compiler-generated drop glue
- any bit pattern (of right length, of course) is correct value of T.
In other words, ByteSequence should represent exactly those types which can be transmuted from/into [u8] safely.

**Comment by valpackett** [comment]
Oh yeah, unsafe trait is a good idea, would be awesome if you implemented that

**Comment by MikailBag** [comment]
I created this crate: https://github.com/MikailBag/as-byte-sequence.
It's my first public project, so I'd like to get your feedback. Does it suit you?

**Comment by valpackett** [comment]
Not sure about an extra crate just for one trait

**Comment by MikailBag** [comment]
I hope such trait can be useful in some other libraries (potentially, for each sockets/networking/messagins/etc library). On my machine, it takes less than a second to compile. Also, tiny-nix-ipc already has quite fat dependencies like nix.
Of course, if you still don't want this new dependency, I can copy this trait to my PR.

**Comment by MikailBag** [comment]
Regarding as-byte-sequence: recently I came across wonderful crate [zerocopy](https://lib.rs/zerocopy).
It seems that traits `AsBytes` and `FromBytes` are perfect fit as trait bounds for `send_struct` and `recv_struct`.
So now my vision looks like following:
- safe methods `send_struct<T: zerocopy::AsBytes>(...)` and `recv_struct<T: zerocopy::FromBytes>(...)`.
- unsafe methods (not sure about naming) `send_struct_unchecked<T>(...)` and `recv_struct_unchecked<T>(...)`.
- unchecked methods are suitable when each side trusts other.
- safe methods should be used otherwise.

Also, `TrustedSocket` type may be introduced, whose API should match current API of `Socket`.
So, when side trust each other, required changes should be small (something like 
`let sock = unsafe {TrustedSocket::new(sock)};`
perhaps)

**Comment by valpackett** [comment]
zerocopy sounds nice. I guess `_raw` would be a better name than `_unchecked`. Not sure about trusted socket.

**Comment by MikailBag** [comment]
I think my PR is ready for your review.

**Comment by valpackett** [comment]
Looks good, thanks!
